src/components/LayerTree/LayerTree.js

```diff
@@ -149,12 +151,31 @@ class LayerTree extends Component {
     if (pos > -1) {
       expandedLayerNames.splice(pos, 1);
     } else {
-      expandedLayerNames.push(layer);
+      expandedLayerNames.push(...this.getExpandedLayers([layer]));
     }
 
     this.setState({ expandedLayerNames });
   }
 
+  /**
+   * Get the always expanded ancestors (isAlwaysExpanded=true) of the given layers
+   * together with the (given) initially expanded layers
+   *
+   * @param  {Layer} layers Initially expanded layers
+   * @return {Array.<Layer>} Initially expanded layers and all its always expanded ancestors
+   */
+  getExpandedLayers(layers) {
+    const { isItemHidden } = this.props;
+    const children = layers.flatMap(l =>
+      l.getChildren().filter(c => !isItemHidden(c) && c.getIsAlwaysExpanded()),
```

Comment:
In addition to comment line 112   should n't test  here also if `!isItemHidden(l) && l.getIsAlwaysExpanded() === true`  ? 

I don't know if it makes sense I just want to avoid to have the same filter in 2 different places.

Reply:
No, if we keep the behavior of https://github.com/geops/react-spatial/pull/537#discussion_r398411227 as the first level is treated differently. So, the given layers are always expanded in this function.